internal/registry/registry_cache.go

```diff
@@ -354,3 +400,38 @@ func (r *Cache) readEntry(ns, name string) (Entry, error) {
 
 	return entry, nil
 }
+
+func (r *Cache) readExtensionEntry(ns, name string) (ExtensionEntry, error) {
```

Comment:
If we do https://github.com/buildpacks/pack/pull/1616/files/48d602bc8cbf9db935a77784fb48fe2186c78bd6..8b9ed1f73ea83b9dbe2ad3aedf08cc84fd4a039d#r1094913127 this could be a shared helper across buildpacks and extensions (passing in `kind` as suggested in https://github.com/buildpacks/pack/pull/1616/files/48d602bc8cbf9db935a77784fb48fe2186c78bd6..8b9ed1f73ea83b9dbe2ad3aedf08cc84fd4a039d#r1094914669)